src/mono/wasm/build/WasmApp.Native.targets

```diff
@@ -712,14 +712,15 @@
     <ILStrip
         Condition=" '$(WasmStripILAfterAOT)' == 'true' "
         TrimIndividualMethods="true"
+        IntermediateOutputPath="$(_WasmIntermediateOutputPath)"
         Assemblies="@(_WasmAssembliesInternal)">
-      <Output TaskParameter="TrimmedAssemblies" ItemName="_ILStripTrimmedAssemblies" />
+      <Output TaskParameter="UpdatedAssemblies" ItemName="_UpdatedAssembliesAfterILStrip" />
     </ILStrip>
-    <Move
-        Condition=" '$(WasmStripILAfterAOT)' == 'true' "
-        SourceFiles="@(_ILStripTrimmedAssemblies->'%(TrimmedAssemblyFileName)')"
-        DestinationFiles="@(_ILStripTrimmedAssemblies)"
-    />
+
+    <ItemGroup>
+      <_WasmAssembliesInternal Remove="@(_WasmAssembliesInternal)" Condition="'$(WasmStripILAfterAOT)' == 'true'" />
+      <_WasmAssembliesInternal Include="@(_UpdatedAssembliesAfterILStrip)" Condition="'$(WasmStripILAfterAOT)' == 'true'" />
```

Comment:
Blazor was never using `@(WasmAssembliesFinal)` and instead rely on `@(ResolvedFileToPublish)`. The AOT compiler works "in place" and so the `@(ResolvedFileToPublish)` contains correct items. We should update `@(ResolvedFileToPublish)` at the end of wasm publish

Reply:
WasmSdk is selecting the files to be bundled here via the `@(WasmAssembliesToBundle)` item which is passed to `WasmApp.targets`:
https://github.com/dotnet/runtime/blob/f2e4ddf20936b37445bbbb2d139e8080ca00f16c/src/mono/nuget/Microsoft.NET.Sdk.WebAssembly.Pack/build/Microsoft.NET.Sdk.WebAssembly.Browser.targets#L595-L600

A corresponding target in Browser.targets should do the "unwrapping", and update whatever items needs to be updated.

Reply:
It didn't have to the unwrapping so just because this assembly location did not changed during AOT, right? 

The flow is that SWA (StaticWebAssets) has dependency on `ProcessPublishFilesForWasm` which triggers `WasmTriggerPublishApp` which triggers `_GatherWasmFilesToPublish` and that copies `@(ResolvedFileToPublish)` to `@(WasmAssembliesToBundle)`. Than neste publish happens and produces `@(WasmAssembilesFinal)`, which has the exact locations as original `@(ResolvedFileToPublish)`. Than finally the `ProcessPublishFilesForWasm` is exected and it uses `@(ResolvedFileToPublish)` to update SWA entries with AOTed assemblies. You suggest to replace the last by reading `@(WasmAssembliesFinal)` instead of `@(ResolvedFileToPublish)`, right?

I'm thinking, why do we need the copying to `WasmAssemblies**`? Why don't we use `ResolvedFileToPublish` for communication between `WasmApp.targets` and `Browser.targets`?

Reply:
> I'm thinking, why do we need the copying to WasmAssemblies**? Why don't we use ResolvedFileToPublish for communication between WasmApp.targets and Browser.targets?

My original plan was to pick up publish artifacts automatically, and figure out "for just bundling", "assemblies", etc. But we also want to move away from `publish` as such, which would be in the sdk first. So, for now using the `WasmAssemblies*` allows us separation from the publish details. Also, `ResolvedFileToPublish` can have metadata to consider once we start adding that.

I also want to drop the nested publish now, since we have better control over what blazor does about the inputs.

Reply:
Do we have an agreement regarding to what should be the right thing to do here?

Reply:
@maraf if you two agreed on the solution. Do you mind pushing a change to this PR?

Reply:
We agreed to use `WasmAssembliesFinal` at the moment, before we drop nested publish. I'll try to update the PR today or at the latest on Monday